app/src/main/java/app/f3d/F3D/android/Utils/FileUtils.java

```diff
@@ -0,0 +1,99 @@
+package app.f3d.F3D.android.Utils;
+
+import android.content.ContentResolver;
+import android.content.Context;
+import android.database.Cursor;
+import android.net.Uri;
+import android.provider.MediaStore;
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+
+public class FileUtils {
+
+    private static String fileExtension;
+
+    public static String createTempFileFromUri(Context context, Uri uri) {
+        try (InputStream inputStream = context.getContentResolver().openInputStream(uri)) {
+//            String mimeType = context.getContentResolver().getType(uri);
+            if (inputStream != null) {
+                String originalFileName = getFileNameFromUri(context.getContentResolver(), uri);
+                String[] nameAndExtension = extractNameAndExtension(originalFileName);
+                fileExtension = nameAndExtension[1];
+                File tempFile = createTempFile(context, nameAndExtension[0], nameAndExtension[1]);
+
+                try (BufferedInputStream bis = new BufferedInputStream(inputStream);
+                     BufferedOutputStream bos = new BufferedOutputStream(Files.newOutputStream(tempFile.toPath()))) {
+
+                    byte[] buffer = new byte[1024];
+                    int bytesRead;
+                    while ((bytesRead = bis.read(buffer)) != -1) {
+                        bos.write(buffer, 0, bytesRead);
+                    }
+                    return tempFile.getAbsolutePath();
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    private static String[] extractNameAndExtension(String fileName) {
+        String[] nameAndExtension = new String[2];
+        String[] parts = fileName.split("\\.");
+
+        if (parts.length >= 2) {
+            nameAndExtension[0] = fileName.substring(0, fileName.lastIndexOf("."));
+            nameAndExtension[1] = parts[parts.length - 1];
+        } else {
+            nameAndExtension[0] = fileName;
```

Comment:
Can it happen? What's the current behavior if the file has no extension?

Reply:
nope, its a redundant code.
removed it, as we are restriciting from a file selection level it needs to have extensions